### PR TITLE
Integrated unfold admin & Added important models fields to display, filter & search

### LINKED ISSUE #6: Unfold admin integration for awesome admin UI and other features
I start working on it but before continue on implementation I need your opinion
Screenshot 2024-02-10 at 14 54 28

[Unfold admin website](https://www.unfoldadmin.com)

### PR DESCRIPTION
**Pull request by M97Chahboun**
fix #6

### CODE REVIEW

**Comment by M97Chahboun** [comment]
@alexeygrigorev 
I suggest to add dashboard for students and teachers

**Comment by M97Chahboun** [comment]
Also I suggest to use WysiwygWidget on course / project / homework description as rich text

**Comment by M97Chahboun** [comment]
for homework questions I removed custom widgets because we have conflict with unfold forms

## Before remove custom widgets
Screenshot 2024-02-11 at 23 29 07

## Before remove custom widgets
Screenshot 2024-02-11 at 23 27 19

File: course_management/settings.py [hunk 1]
```diff
@@ -109,12 +111,17 @@
...
-DATABASE_URL = os.getenv("DATABASE_URL", "sqlite:///db/db.sqlite3")
-db_config = dj_database_url.config(default=DATABASE_URL)
-
-DATABASES = {"default": db_config}
+# DATABASE_URL = os.getenv("DATABASE_URL", "sqlite:///db/db.sqlite3")
+# db_config = dj_database_url.config(default=DATABASE_URL)
 
+# DATABASES = {"default": db_config}
 
+DATABASES = {
```
Comment (alexeygrigorev) [inline course_management/settings.py:119]: Can you please unroll this change? To run it locally with sqlite, use `export DATABASE_URL=sqlite:///db/db.sqlite3`

**Comment by alexeygrigorev** [comment]
Is there a way to adjust the width of the widgets? e.g. I want to have wider answers widget and narrower scores widget

**Comment by alexeygrigorev** [comment]
we probably also need to add unfold to pipenv?

File: course_management/settings.py (hunk 1 above)
Comment (M97Chahboun) [reply course_management/settings.py:119]: Sorry, I will revert it.

**Comment by M97Chahboun** [comment]
is better ?

Screenshot 2024-02-13 at 22 21 01

**Comment by alexeygrigorev** [comment]
Yes, looks very nice! Can you please resolve the conflict and then I'll merge it

File: accounts/admin.py [hunk 2]
```diff
@@ -1,3 +1,62 @@
 from django.contrib import admin
+from unfold.admin import ModelAdmin
+from django.contrib.auth.models import Group
 
 # Register your models here.
+from allauth.account.models import EmailAddress
```
Comment (alexeygrigorev) [inline accounts/admin.py:6]: Also I think this isn't really needed, let's revert this one, just for courses is enough

Comment (M97Chahboun) [reply accounts/admin.py:6]: This added for adapt email adress admin with unfold styling 

when remove it 
result: 
Screenshot 2024-02-15 at 15 18 00

**Comment by alexeygrigorev** [comment]
what are the styles? Is it possible not to add them? I thought the styles would be fetched from the app static files

File: course_management/settings.py [hunk 3]
```diff
@@ -149,13 +152,12 @@
 # Static files (CSS, JavaScript, Images)
 # https://docs.djangoproject.com/en/4.2/howto/static-files/
 
-STATIC_ROOT = BASE_DIR / "static"
+STATIC_ROOT = BASE_DIR / "staticfiles"
```
Comment (alexeygrigorev) [inline course_management/settings.py:155]: please revert it

**Comment by alexeygrigorev** [comment]
the PR is getting bigger and bigger. To be honest, I'm now hesitant to accept it - can we please make fewer changes like a few commits ago?

File: accounts/admin.py (hunk 2 above)
Comment (alexeygrigorev) [reply accounts/admin.py:6]: Why do we need a panel for it? I was hoping we could remove the entire accounts/admin.py file

File: course_management/settings.py [hunk 4]
```diff
@@ -149,13 +152,12 @@
...
 STATIC_URL = "static/"
 
-# STATICFILES_DIRS = [
-#     STATIC_ROOT,
-#     # Other directories here
-# ]
+STATICFILES_DIRS = [
```
Comment (alexeygrigorev) [inline course_management/settings.py:158]: why do we need it?

File: courses/admin.py [hunk 5]
```diff
@@ -16,16 +53,15 @@ class QuestionForm(forms.ModelForm):
...
-            ),
-            "correct_answer": forms.TextInput(attrs={"size": "20"}),
+            "text": forms.TextInput(attrs={"size": "60", 'class': field_styles}),
+            "possible_answers": forms.Textarea(attrs={"cols": 60, "rows": 4, 'class': field_styles}),
+            "correct_answer": forms.TextInput(attrs={"size": "20", 'class': field_styles}),
         }
-
-
-class QuestionInline(admin.TabularInline):  # or admin.StackedInline
+
```
Comment (alexeygrigorev) [inline courses/admin.py:62]: please remove

File: courses/admin.py [hunk 6]
```diff
@@ -35,13 +71,9 @@ def score_selected_homeworks(modeladmin, request, queryset):
     for homework in queryset:
         status, message = score_homework_submissions(homework.id)
         if status:
-            modeladmin.message_user(
-                request, message, level=messages.SUCCESS
-            )
+            modeladmin.message_user(request, message, level=messages.SUCCESS)
```
Comment (alexeygrigorev) [inline courses/admin.py:74]: please undo the format changes

File: courses/admin.py [hunk 7]
```diff
@@ -35,13 +71,9 @@ def score_selected_homeworks(modeladmin, request, queryset):
...
         else:
-            modeladmin.message_user(
-                request, message, level=messages.WARNING
-            )
+            modeladmin.message_user(request, message, level=messages.WARNING)
```
Comment (alexeygrigorev) [inline courses/admin.py:76]: and here too

File: accounts/admin.py (hunk 2 above)
Comment (M97Chahboun) [reply accounts/admin.py:6]: for update email address admin styling for to be as unfold styling...

File: course_management/settings.py (hunk 4 above)
Comment (M97Chahboun) [reply course_management/settings.py:158]: I have some issues with tailwind styling when update questions but now is well

**Comment by M97Chahboun** [comment]
@alexeygrigorev 
Im sorry I have some issues with tailwind when try to update questions form but I think now 
is better

File: accounts/admin.py (hunk 2 above)
Comment (M97Chahboun) [reply accounts/admin.py:6]: if you want hide email address and other models as (sites and socialaccount) models let me know we can unregister this models from admin 
but if you want show it we should [re-register](https://unfoldadmin.com/installation/) with unfold for apply styling

**Comment by M97Chahboun** [comment]
@alexeygrigorev

**Comment by alexeygrigorev** [comment]
Thank you!

**Comment by alexeygrigorev** [comment]
image
looks really great

**Pull request merged**